internal/toolexec/aspect/oncompile-main.go

```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package aspect
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/datadog/orchestrion/internal/log"
+	"github.com/datadog/orchestrion/internal/toolexec/aspect/linkdeps"
+	"github.com/datadog/orchestrion/internal/toolexec/importcfg"
+	"github.com/datadog/orchestrion/internal/toolexec/proxy"
+	"github.com/dave/jennifer/jen"
+)
+
+// OnCompileMain only performs changes when compiling the "main" package, adding blank imports for
+// any linkdeps dependencies that are not yet satisfied by the importcfg file. This ensures that the
+// relevant packages' `init` (if any) are appropriately run, and that the linker automatically picks
+// up these dependencies when creating the full binary.
```

Comment:
I don't fully agree with this part of the comment: up until now our instrumentations were properly importing the packages they required themselves. But now that you leveraged go:linkname alone, indeed you no longer import the package defining it. So I'd write the rational in this comment around this problem instead.